server/src/main/java/org/elasticsearch/cluster/metadata/IndexReshardingMetadata.java

```diff
@@ -207,15 +208,22 @@ public static IndexReshardingMetadata newSplitByMultiple(int shardCount, int mul
         return new IndexReshardingMetadata(IndexReshardingState.Split.newSplitByMultiple(shardCount, multiple));
     }
 
-    public IndexReshardingMetadata transitionSplitTargetToHandoff(ShardId shardId) {
+    public static boolean isSplitTarget(ShardId shardId, @Nullable IndexReshardingMetadata reshardingMetadata) {
+        return reshardingMetadata != null && reshardingMetadata.isSplit() && reshardingMetadata.getSplit().isTargetShard(shardId.id());
+    }
+
+    public IndexReshardingMetadata transitionSplitTargetToNewState(
+        ShardId shardId,
+        IndexReshardingState.Split.TargetShardState newTargetState
+    ) {
         assert state instanceof IndexReshardingState.Split;
         IndexReshardingState.Split splitState = (IndexReshardingState.Split) state;
         IndexReshardingState.Split.TargetShardState[] newTargets = Arrays.copyOf(
             splitState.targetShards(),
             splitState.targetShards().length
         );
         int i = shardId.getId() - state.shardCountBefore();
-        newTargets[i] = IndexReshardingState.Split.TargetShardState.HANDOFF;
+        newTargets[i] = newTargetState;
```

Comment:
I think based on the split state machine it's illegal and dangerous to jump from a state to any other arbitrary state, so it might be good to at least add assertions here that the new state is legal from the current state.